pyproject.toml

```diff
@@ -74,6 +74,7 @@ jsonpath-ng = {optional = true, version = "^1.6.0"}
 rank-bm25 = {optional = true, version = "^0.2.2"}
 scikit-learn = {optional = true, version = "*"}
 spacy = {optional = true, version = "^3.7.1"}
+tritonclient = {optional = true, version = ">=2.39.0"}
```

Comment:
do we want to lazy import this pkg instead of adding a dep pkg

Reply:
@hatianzhang meaning remove the tritonclient dependency and instead rely on the user to manually install it if desired?

Reply:
yea tbh, that might be ideal (still figuring out how we want to manage these poetry dependencies)

Raising an error if not installed should be ok :)